### ISSUE TITLE
polars.LazyFrame.profile(show_plot=true) not working 

### ISSUE THREAD

**Issue by romanovacca-hema** [pola-rs/polars#6492]
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Issue description

When running the example code for [.profile()](https://pola-rs.github.io/polars/py-polars/html/reference/lazyframe/api/polars.LazyFrame.profile.html#polars.LazyFrame.profile) , with parameter "show_plot" = True, it gives the following error:
`notImplementerError: Conversion of polars data type Utf8 to C-type not implemented when running example code`

So in the example, the column `a` = utf8  the error seems to point to the dtype, which is confusing since the `show_plot` should plot the values of the steps of the query, not the column values.

![Screenshot 2023-01-27 at 13 48 43](https://user-images.githubusercontent.com/98817897/215090846-a60d6771-2d12-430b-b2ef-0d9615bd97e6.png)

When running this function with the `show_plot` set to false, the .profile() works fine.

### Reproducible example

```python
df = pl.DataFrame(
    {
        "a": ["a", "b", "a", "b", "b", "c"],
        "b": [1, 2, 3, 4, 5, 6],
        "c": [6, 5, 4, 3, 2, 1],
    }
).lazy()
df.groupby("a", maintain_order=True).agg(pl.all().sum()).sort(
    "a"
).profile(show_plot=True)
```

### Expected behavior

This should show the graph without problems

### Installed versions

<details>

```
>>> pl.show_versions()
---Version info---
Polars: 0.15.18
Index type: UInt32
Platform: macOS-10.16-x86_64-i386-64bit
Python: 3.10.9 (main, Jan 11 2023, 09:18:20) [Clang 14.0.6 ]
---Optional dependencies---
pyarrow: <not installed>
pandas: <not installed>
numpy: 1.24.1
fsspec: <not installed>
connectorx: <not installed>
xlsx2csv: <not installed>
deltalake: <not installed>
matplotlib: 3.6.3```

</details>

**Comment by ritchie46** [comment]
That's strange. I cannot replicate this:

![image](https://user-images.githubusercontent.com/3023000/215094510-74e7f6f2-ca5c-4056-8c56-ff07e57bd8c1.png)

**Comment by romanovacca** [comment]
That is weird. I get this:
![Screenshot 2023-01-27 at 15 57 47](https://user-images.githubusercontent.com/12878583/215117091-d386e328-b2ac-442f-b595-536332993cb9.png)

I have added the traceback below. Is there another way I can provide this more clear, then please let me know @ritchie46 !

<details>
  <summary>traceback</summary>
`NotImplementedError                       Traceback (most recent call last)
Cell In[2], line 8
      1 df = pl.DataFrame(
      2     {
      3         "a": ["a", "b", "a", "b", "b", "c"],
   (...)
      6     }
      7 ).lazy()
----> 8 df.groupby("a", maintain_order=True).agg(pl.all().sum()).sort(
      9     "a"
     10 ).profile(show_plot=True)

File ~/opt/anaconda3/envs/polars_env/lib/python3.10/site-packages/polars/utils.py:393, in deprecated_alias.<locals>.deco.<locals>.wrapper(*args, **kwargs)
    390 @functools.wraps(fn)
    391 def wrapper(*args: P.args, **kwargs: P.kwargs) -> T:
    392     _rename_kwargs(fn.__name__, kwargs, aliases)
--> 393     return fn(*args, **kwargs)

File ~/opt/anaconda3/envs/polars_env/lib/python3.10/site-packages/polars/internals/lazyframe/frame.py:1082, in LazyFrame.profile(self, type_coercion, predicate_pushdown, projection_pushdown, simplify_expression, no_optimization, slice_pushdown, common_subplan_elimination, show_plot, truncate_nodes, figsize, streaming)
   1077     timings_ = timings_.with_columns(
   1078         pli.col("node").str.slice(0, truncate_nodes) + "..."
   1079     )
   1081 max_in_unit = timings_["end"][0]
-> 1082 ax.barh(
   1083     timings_["node"],
   1084     width=timings_["end"] - timings_["start"],
   1085     left=timings_["start"],
   1086 )
   1088 plt.title("Profiling result")
   1089 ax.set_xlabel(f"node duration in [{unit}], total {max_in_unit}{unit}")

File ~/opt/anaconda3/envs/polars_env/lib/python3.10/site-packages/matplotlib/axes/_axes.py:2629, in Axes.barh(self, y, width, height, left, align, data, **kwargs)
   2519 r"""
   2520 Make a horizontal bar plot.
   2521 
   (...)
   2626 :doc:`/gallery/lines_bars_and_markers/horizontal_barchart_distribution`.
   2627 """
   2628 kwargs.setdefault('orientation', 'horizontal')
-> 2629 patches = self.bar(x=left, height=height, width=width, bottom=y,
   2630                    align=align, data=data, **kwargs)
   2631 return patches

File ~/opt/anaconda3/envs/polars_env/lib/python3.10/site-packages/matplotlib/__init__.py:1433, in _preprocess_data.<locals>.inner(ax, data, *args, **kwargs)
   1430 @functools.wraps(func)
   1431 def inner(ax, *args, data=None, **kwargs):
   1432     if data is None:
-> 1433         return func(ax, *map(sanitize_sequence, args), **kwargs)
   1435     bound = new_sig.bind(ax, *args, **kwargs)
   1436     auto_label = (bound.arguments.get(label_namer)
   1437                   or bound.kwargs.get(label_namer))

File ~/opt/anaconda3/envs/polars_env/lib/python3.10/site-packages/matplotlib/axes/_axes.py:2377, in Axes.bar(self, x, height, width, bottom, align, **kwargs)
   2375         self.set_yscale('log', nonpositive='clip')
   2376 else:  # horizontal
-> 2377     self._process_unit_info(
   2378         [("x", width), ("y", y)], kwargs, convert=False)
   2379     if log:
   2380         self.set_xscale('log', nonpositive='clip')

File ~/opt/anaconda3/envs/polars_env/lib/python3.10/site-packages/matplotlib/axes/_base.py:2578, in _AxesBase._process_unit_info(self, datasets, kwargs, convert)
   2576     # Update from data if axis is already set but no unit is set yet.
   2577     if axis is not None and data is not None and not axis.have_units():
-> 2578         axis.update_units(data)
   2579 for axis_name, axis in axis_map.items():
   2580     # Return if no axis is set.
   2581     if axis is None:

File ~/opt/anaconda3/envs/polars_env/lib/python3.10/site-packages/matplotlib/axis.py:1602, in Axis.update_units(self, data)
   1596 def update_units(self, data):
   1597     """
   1598     Introspect *data* for units converter and update the
   1599     ``axis.converter`` instance if necessary. Return *True*
   1600     if *data* is registered for unit conversion.
   1601     """
-> 1602     converter = munits.registry.get_converter(data)
   1603     if converter is None:
   1604         return False

File ~/opt/anaconda3/envs/polars_env/lib/python3.10/site-packages/matplotlib/units.py:184, in Registry.get_converter(self, x)
    182 """Get the converter interface instance for *x*, or None."""
    183 # Unpack in case of e.g. Pandas or xarray object
--> 184 x = cbook._unpack_to_numpy(x)
    186 if isinstance(x, np.ndarray):
    187     # In case x in a masked array, access the underlying data (only its
    188     # type matters).  If x is a regular ndarray, getdata() just returns
    189     # the array itself.
    190     x = np.ma.getdata(x).ravel()

File ~/opt/anaconda3/envs/polars_env/lib/python3.10/site-packages/matplotlib/cbook/__init__.py:2331, in _unpack_to_numpy(x)
   2328     return x
   2329 if hasattr(x, 'to_numpy'):
   2330     # Assume that any function to_numpy() do actually return a numpy array
-> 2331     return x.to_numpy()
   2332 if hasattr(x, 'values'):
   2333     xtmp = x.values

File ~/opt/anaconda3/envs/polars_env/lib/python3.10/site-packages/polars/internals/series/series.py:2822, in Series.to_numpy(self, zero_copy_only, writable, *args)
   2820         np_array = convert_to_date(self.view(ignore_nulls=True))
   2821     else:
-> 2822         np_array = self.view(ignore_nulls=True)
   2823 elif self.is_datelike():
   2824     np_array = convert_to_date(self._s.to_numpy())

File ~/opt/anaconda3/envs/polars_env/lib/python3.10/site-packages/polars/internals/series/series.py:2758, in Series.view(self, ignore_nulls)
   2754     assert not self.has_validity()
   2756 from polars.internals.series._numpy import SeriesView, _ptr_to_numpy
-> 2758 ptr_type = dtype_to_ctype(self.dtype)
   2759 ptr = self._s.as_single_ptr()
   2760 array = _ptr_to_numpy(ptr, self.len(), ptr_type)

File ~/opt/anaconda3/envs/polars_env/lib/python3.10/site-packages/polars/datatypes.py:661, in dtype_to_ctype(dtype)
    659     return DataTypeMappings.DTYPE_TO_CTYPE[dtype]
    660 except KeyError:  # pragma: no cover
--> 661     raise NotImplementedError(
    662         f"Conversion of polars data type {dtype} to C-type not implemented."
    663     ) from None

NotImplementedError: Conversion of polars data type Utf8 to C-type not implemented.`
  </details>

**Comment by ritchie46** [comment]
Anyone with a macbook that can confirm a reproduction?

**Comment by KishManani** [comment]
> Anyone with a macbook that can confirm a reproduction?

I'm able to run this successfully on a macbook (not using the M1 chip) with polars 0.15.18.

<img width="1005" alt="image" src="https://user-images.githubusercontent.com/30973056/215258516-564ca806-2b58-47ac-9138-3b1ace447a63.png">

**Comment by romanovacca** [comment]
Ran this on someone else's M1 macbook and that resulted in exactly the same issue

**Comment by romanovacca** [comment]
@ritchie46 mentioned this is most likely a numpy issue, so I will have a look at this.

**Comment by romanovacca** [comment]
I looked into this and found that the error had to do with the fact that there is no c-type defined for utf8.
But when looking through the code I saw this note in the .to_numpy() function:

` If you are attempting to convert Utf8 to an array you'll need to install pyarrow.`

So after installing pyarrow, the code above actually works and it had nothing to do with M1 chip macbook :)

So how I see it there are two options:
1: Add a warning to the user that the pyarrow library is necessary when show_graph = True (In the same function we actually already check to see if pyarrow is installed, but we fail to give the right feedback in this specific scenario.
)

2: we could extend the function `DTYPE_TO_CTYPE` with I think `Utf8: ctypes.c_char_p,` , but then we run into the following:
`ptr_type = dtype_to_ctype(self.dtype)`
 `ptr = self._s.as_single_ptr()`
Where the last function is only implemented for numeric types.

What is the preferred solution @ritchie46 ?
